Source/drlg_l1.cpp

```diff
@@ -830,45 +758,37 @@ void PlaceMiniSetRandom(const BYTE *miniset, int rndper)
 					if (miniset[ii] != 0 && dungeon[xx + sx][yy + sy] != miniset[ii]) {
 						found = false;
 					}
-					// BUGFIX: Check if tile is Protected (fixing this breaks crypt levels)
+					// BUGFIX: Check if tile is Protected (fixing this breaks compatability)
 					ii++;
 				}
 			}
+			if (!found)
+				continue;
 			int kk = sw * sh + 2;
-			if (miniset[kk] >= 84 && miniset[kk] <= 100 && found) {
-				// BUGFIX: accesses to dungeon can go out of bounds (fixed)
-				// BUGFIX: Comparisons vs 100 should use same tile as comparisons vs 84 (fixing this breaks crypt levels)
-
-				constexpr auto ComparisonWithBoundsCheck = [](Point p1, Point p2) {
-					return (p1.x >= 0 && p1.x < DMAXX && p1.y >= 0 && p1.y < DMAXY) && (p2.x >= 0 && p2.x < DMAXX && p2.y >= 0 && p2.y < DMAXY) && (dungeon[p1.x][p1.y] >= 84 && dungeon[p2.x][p2.y] <= 100);
-				};
-				if (ComparisonWithBoundsCheck({ sx - 1, sy }, { sx - 1, sy })) {
-					found = false;
-				}
-				if (ComparisonWithBoundsCheck({ sx + 1, sy }, { sx - 1, sy })) {
-					found = false;
-				}
-				if (ComparisonWithBoundsCheck({ sx, sy + 1 }, { sx - 1, sy })) {
-					found = false;
-				}
-				if (ComparisonWithBoundsCheck({ sx, sy - 1 }, { sx - 1, sy })) {
-					found = false;
-				}
-			}
-			if (found && GenerateRnd(100) < rndper) {
-				for (int yy = 0; yy < sh; yy++) {
-					for (int xx = 0; xx < sw; xx++) {
-						if (miniset[kk] != 0) {
-							dungeon[xx + sx][yy + sy] = miniset[kk];
-						}
-						kk++;
+			if (!CanReplaceTile(miniset[kk], { sx, sy }))
+				continue;
+			if (GenerateRnd(100) >= rndper)
+				continue;
+			for (int yy = 0; yy < sh; yy++) {
+				for (int xx = 0; xx < sw; xx++) {
+					if (miniset[kk] != 0) {
+						dungeon[xx + sx][yy + sy] = miniset[kk];
 					}
+					kk++;
 				}
 			}
 		}
 	}
 }
 
+void PlaceMiniSetRandom1x1(uint8_t search, uint8_t replace, int rndper)
+{
+	uint8_t miniSet[4] { 1, 1, 0, 0 };
+	miniSet[2] = search;
+	miniSet[3] = replace;

```

Comment:
```suggestion
	uint8_t miniSet[4] { 1, 1, search, replace };
```
should be supported? Is there a compiler (gcc 6?) that doesn't allow this?